### PR TITLE
User can update their profile picture by selecting an image from their photo gallery

### PR DESCRIPTION
**Pull request by emmymarysia**
Within the fragment_profile layout, I added a button for the user to select an image for their profile picture from the phone's photo gallery. Within the ProfileFragment Java code, I set the onClickListener for this button to call a method that launches an external intent that opens the gallery and allows the user to pick a photo. I then added the code to get back the data for the photo they selected within the onActivityResult method. I also created a helper method called saveAndShowImage that takes in a bitmap and uploads the image to Parse/displays it on the profile screen regardless of whether it was taken by the camera or chosen from the gallery. To test so far I have used the emulator and verified that the image I selected matches what is uploaded to Parse.

### CODE REVIEW

**Comment by katzjm** [comment]
A lot of this description is describing what the code is. Focus more on the functionality, or the user experience. 

My example based on reading: "Adds a button that, when clicked, opens the gallery for the user to pick a photo. This works if it's a new photo or from the gallery, and saves it to Parse"

File: app/src/main/java/com/example/golocal/fragments/ProfileFragment.java [hunk 1]
```diff
@@ -48,12 +52,14 @@ public class ProfileFragment extends Fragment {
...
     private Button btFinishEditBio;
     private ImageButton ibEditProfileImage;
+    private Button btChooseFromGallery;
     private ParseUser user;
     private final String KEY_BIO = "bio";
     private final String KEY_IMAGE = "profileImage";
     private Context context;
     private final String TAG = "ProfileFragment";
     public static final int CAPTURE_IMAGE_ACTIVITY_REQUEST_CODE = 42;
+    public final static int PICK_PHOTO_CODE = 1046;
```
Comment (katzjm) [inline app/src/main/java/com/example/golocal/fragments/ProfileFragment.java:62]:
```suggestion
    public static final int PICK_PHOTO_CODE = 1046;
```
Also, why 1046 and why public?

File: app/src/main/java/com/example/golocal/fragments/ProfileFragment.java [hunk 2]
```diff
@@ -181,27 +195,67 @@ public void onActivityResult(int requestCode, int resultCode, @Nullable Intent d
...
-                            Toast.makeText(context, "Error while saving profile image", Toast.LENGTH_SHORT).show();
-                        }
-                    }
-                });
-                Glide.with(this)
-                        .load(user.getParseFile(KEY_IMAGE).getUrl())
-                        .override(Target.SIZE_ORIGINAL, Target.SIZE_ORIGINAL)
-                        .circleCrop()
-                        .into(ivProfileImage);
+                saveAndShowImage(resizedBitmap);
```
Comment (katzjm) [inline app/src/main/java/com/example/golocal/fragments/ProfileFragment.java:199]: Good job refactoring to keep code DRY (Don't Repeat Yourself)!

File: app/src/main/java/com/example/golocal/fragments/ProfileFragment.java [hunk 3]
```diff
@@ -181,27 +195,67 @@ public void onActivityResult(int requestCode, int resultCode, @Nullable Intent d
...
+                .override(Target.SIZE_ORIGINAL, Target.SIZE_ORIGINAL)
+                .circleCrop()
+                .into(ivProfileImage);
+    }
+
+    public Bitmap loadFromUri(Uri photoUri) {
+        Bitmap image = null;
+        try {
+            // check version of Android on device
+            if(Build.VERSION.SDK_INT > 27){
```
Comment (katzjm) [inline app/src/main/java/com/example/golocal/fragments/ProfileFragment.java:235]:
```suggestion
            if (Build.VERSION.SDK_INT > 27){
```

File: app/src/main/java/com/example/golocal/fragments/ProfileFragment.java [hunk 4]
```diff
@@ -181,27 +195,67 @@ public void onActivityResult(int requestCode, int resultCode, @Nullable Intent d
...
+                // on newer versions of Android, use the new decodeBitmap method
+                ImageDecoder.Source source = ImageDecoder.createSource(context.getContentResolver(), photoUri);
+                image = ImageDecoder.decodeBitmap(source);
+            } else {
+                // support older versions of Android by using getBitmap
+                image = MediaStore.Images.Media.getBitmap(context.getContentResolver(), photoUri);
```
Comment (katzjm) [inline app/src/main/java/com/example/golocal/fragments/ProfileFragment.java:241]: Nice job handling edge cases

File: app/src/main/java/com/example/golocal/fragments/ProfileFragment.java (hunk 1 above)
Comment (katzjm) [reply app/src/main/java/com/example/golocal/fragments/ProfileFragment.java:62]: In that case, could we make it `private`?

**Review by katzjm — APPROVED** [review]

File: app/src/main/java/com/example/golocal/fragments/ProfileFragment.java (hunk 1 above)
Comment (Anish2607) [reply app/src/main/java/com/example/golocal/fragments/ProfileFragment.java:62]: In cases like this, we can add a comment in code to just explain that this is a random number

File: app/src/main/java/com/example/golocal/fragments/ProfileFragment.java [hunk 5]
```diff
@@ -181,27 +195,67 @@ public void onActivityResult(int requestCode, int resultCode, @Nullable Intent d
...
+        ByteArrayOutputStream stream = new ByteArrayOutputStream();
+        image.compress(Bitmap.CompressFormat.JPEG, 100, stream);
+        byte[] bitmapBytes = stream.toByteArray();
+        user.put(KEY_IMAGE, new ParseFile(bitmapBytes));
+        user.saveInBackground(new SaveCallback() {
+            @Override
+            public void done(ParseException e) {
+                if (e != null) {
+                    Log.e(TAG, "Error while saving profile image", e);
+                    Toast.makeText(context, "Error while saving profile image", Toast.LENGTH_SHORT).show();
```
Comment (Anish2607) [inline app/src/main/java/com/example/golocal/fragments/ProfileFragment.java:220]: Save string to resource file

File: app/src/main/java/com/example/golocal/fragments/ProfileFragment.java [hunk 6]
```diff
@@ -181,27 +195,67 @@ public void onActivityResult(int requestCode, int resultCode, @Nullable Intent d
...
+                }
+            }
+        });
+        Glide.with(this)
```
Comment (Anish2607) [inline app/src/main/java/com/example/golocal/fragments/ProfileFragment.java:224]: We can also show a toast to let the user know that the picture was updated successfully.

**Pull request merged**